File: golang/src/diadust/models/query.go
package models

const (
	ORDER_TYPE_CREATED    = "CREATED"
	ORDER_TYPE_SIZE       = "SIZE"
	ORDER_TYPE_RESOLUTION = "RESOLUTION"
	ORDER_DESC            = "DESC"
	ORDER_ASC             = "ASC"
)

type ImageSearchFilter struct {
	IncludeNames []string `json:"includeNames"`
	ExcludeNames []string `json:"excludeNames"`
	IncludeTags  []string `json:"includeTags"`
	ExcludeTags  []string `json:"excludeTags"`
	OrderType    string   `json:"orderType"`
	Order        string   `json:"order"`
}

File: electron/message/model/Query.ts
export interface ImageSearchFilter {
    includeNames?: Array<string>;   // search by names
    excludeNames?: Array<string>;   // search result shall not include names
    includeTags?: Array<number>;    // search by tags
    excludeTags?: Array<number>;    // search result shall not include tags
    orderType: FilterOrderType;     // order type
    order: FilterOrder;             // order
}

enum FilterOrderType {
    CREATED,    // record created
    SIZE,       // file size
    RESOLUTION, // image resolution
}

enum FilterOrder {
    DESC,
    ASC,
}